src/activity_actor.cpp

```diff
@@ -1140,6 +1140,7 @@ void consume_activity_actor::finish( player_activity &act, Character & )
     if( !canceled ) {
         if( consume_location ) {
             player_character.consume( consume_location, /*force=*/true );
```

Comment:
It seems there's code in `player::consume` to handle liquid spilling, but it is currently not working.

Reply:
Dang, I should have looked into that method. I'll take a closer at it look tomorrow. Seems a bit of redundancy has creeped in.